File: Java/BaseClasses/Coordinates.java
package Second_sem.lab5.Java.BaseClasses;

import java.util.ArrayList;

public class Coordinates {
    private double x; //Максимальное значение поля: 36
    private Float y; //Значение поля должно быть больше -821, Поле не может быть null

    @Override
    public String toString() {
        return "["+x+"; "+y+"]";
    }

    public Coordinates(double x, Float y){
        this.x = x;
        this.y = y;
    }
    public Coordinates(ArrayList<Number> array){
        this.x=(double) array.get(0);
        Double unzipY = (Double) array.get(1);
        this.y= unzipY.floatValue();
    }
}

File: Kotlin/BaseClasses/Coordinates.kt
package Second_sem.lab5.Kotlin.BaseClasses

class Coordinates (private val x: Double = 0.0, private val y: Float = 0.0F){
    override fun toString() : String{
        return "[$x; $y]"
    }
    constructor(list: List<Number>) :this(list[0] as Double, list[1].toFloat())
}